ash/src/entry.rs

```diff
@@ -153,24 +153,40 @@ impl EntryV1_0 for Entry<V1_0> {
     }
 }
 
-impl<V: FunctionPointers> Entry<V> {
-    pub fn new() -> Result<Self, LoadingError> {
-        let lib = VK_LIB
-            .as_ref()
-            .map_err(|err| LoadingError::LibraryLoadError(err.clone()))?;
-
-        let static_fn = vk::StaticFn::load(|name| unsafe {
-            lib.symbol(&*name.to_string_lossy())
-                .unwrap_or(ptr::null_mut())
-        }).map_err(LoadingError::StaticLoadError)?;
+impl<V: FunctionPointers, L> EntryCustom<V, L> {
+    pub fn new_custom<OF, LF>(o: OF, mut l: LF) -> Result<Self, LoadingError>
+    where
+        OF: FnOnce() -> Result<L, LoadingError>,
```

Comment:
Could you give the generics more explicit names? Maybe something like `Open` and `Load` instead of `OF`and `LF`